compiler/rustc_metadata/src/rmeta/mod.rs

```diff
@@ -41,6 +41,10 @@ crate fn rustc_version() -> String {
     format!("rustc {}", option_env!("CFG_VERSION").unwrap_or("unknown version"))
 }
 
+crate fn crate_svh(tcx: TyCtxt<'_>) -> String {
+    tcx.crate_hash(LOCAL_CRATE).to_string()
+}
+
 /// Metadata encoding version.
 /// N.B., increment this if you change the format of metadata such that
 /// the rustc version can't be found to compare with `rustc_version()`.
```

Comment:
Does it - I don't believe this is a breaking change? We have a pointer to the start which before pointed to after the rust version, but now points to after the rust version and svh. The previous code will just ignore the svh as it isn't looking to read any further than the rust version before it then jumps to the base location for the main metadata.

Reply:
Ah you're right because otherwise you'd have to do a little deduction to figure out when reading the metadata that this was a version of the compiler that had included a svh or not.

Reply:
I'm going to need a hint. When I change the metadata version the previous compiler won't compile the new one. There's got to be a trick I can use to bootstrap?